server/src/main/java/org/candlepin/model/OwnerInfoBuilder.java

```diff
@@ -0,0 +1,119 @@
+import org.candlepin.dto.api.v1.OwnerInfo;
+
+import java.util.List;
+
+/**
+ * OwnerInfoBuilder is used to accumulate data to build OwnerInfo with dynamic values.
+ */
+
+public class OwnerInfoBuilder {
+
+    private OwnerInfo ownerInfo;
+
+    public static final String PHYSICAL = "physical";
+    public static final String GUEST = "guest";
+
+    public OwnerInfoBuilder() {
+        ownerInfo = new OwnerInfo();
+    }
+
+    public OwnerInfo build() {
+        return this.ownerInfo;
+    }
+
+    public void addTypeTotal(ConsumerType type, int consumers, int entitlements) {
+        ownerInfo.putConsumerCounts(type.getLabel(), consumers);
+        ownerInfo.putEntitlementsConsumedByType(type.getLabel(), entitlements);
+    }
+
+    public void addToConsumerTypeCountByPool(ConsumerType type, int toAdd) {
+        Integer count = ownerInfo.getConsumerTypeCountByPool().get(type.getLabel());
+        if (count == null) {
+            count = 0;
+        }
+        ownerInfo.putConsumerTypeCountByPool(type.getLabel(), count + toAdd);
+    }
+
+    public void addToEnabledConsumerTypeCountByPool(ConsumerType type, int toAdd) {
+        Integer count = ownerInfo.getEnabledConsumerTypeCountByPool().get(type.getLabel());
+        if (count == null) {
+            count = 0;
+        }
+        ownerInfo.putEnabledConsumerTypeCountByPool(type.getLabel(), count + toAdd);
+    }
+
+    public void setConsumerTypesByPool(List<ConsumerType> consumerTypes) {
+        for (ConsumerType c : consumerTypes) {
+            ownerInfo.putConsumerTypeCountByPool(c.getLabel(), 0);
+        }
+    }
+
+    public void addToEntitlementsConsumedByFamily(String family, int physical,
+        int virtual) {
+        ConsumptionTypeCountsDTO typeCounts;
+        if (!ownerInfo.getEntitlementsConsumedByFamily().containsKey(family)) {
+            typeCounts = new ConsumptionTypeCountsDTO().physical(0).guest(0);
+            ownerInfo.putEntitlementsConsumedByFamily(family, typeCounts);
+        }
+        else {
+            typeCounts = ownerInfo.getEntitlementsConsumedByFamily().get(family);
+        }
+
+        typeCounts.setPhysical(typeCounts.getPhysical() + physical);
+        typeCounts.setGuest(typeCounts.getGuest() + virtual);
+    }
+
+    public void addDefaultEntitlementsConsumedByFamily(int physical, int virtual) {
+        for (String key : ownerInfo.getEntitlementsConsumedByFamily().keySet()) {
+            ConsumptionTypeCountsDTO count = ownerInfo.getEntitlementsConsumedByFamily().get(key);
+            physical -= count.getPhysical();
+            virtual -= count.getGuest();
+        }
+
+        // just ignore the default family if we have nothing to put in it.
+        if (physical > 0 || virtual > 0) {
+            addToEntitlementsConsumedByFamily("none", physical, virtual);
+        }
+    }
+
+    public void addDefaultEnabledConsumerTypeCount(int activePools) {
+        for (String key : ownerInfo.getEntitlementsConsumedByType().keySet()) {
```

Comment:
This should be iterating over `getConsumerTypeCountByPool`

Reply:
Corrected.